File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

	boolean isArmstrongNumber(int numberToCheck) {
	    String number = String.valueOf(numberToCheck);
	    int numberOfDigits = number.length();

	    if (numberOfDigits == 1) {
	    	return true;
		}

		int divider = (int)Math.pow(10, numberOfDigits - 1);
	    int reminder = numberToCheck;
	    int amstrongNumber = 0;
	    while (reminder != 0) {
	        int digit = reminder / divider;
	        reminder %= divider;
	        amstrongNumber += (int)Math.pow(digit, numberOfDigits);
	        divider /= 10;
        }

		if (amstrongNumber == numberToCheck) {
		    return true;
        }

        return false;
	}
}

File: csharp/armstrong-numbers/ArmstrongNumbers.cs
using System;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        int original = number;
        var length = number.ToString().Length;
        var calculated = 0;
        var divider = (int)Math.Pow(10, (length - 1));
        while (number != 0)
        {
            calculated += (int)Math.Pow(number / divider, length);
            number %= divider;
            divider /= 10;
        }

        return original == calculated;
    }
}